### ISSUE TITLE
TypeError: Can't read property 'bonds' of null

### ISSUE THREAD

**Issue by dxdc** [aarons22/homebridge-bond#39]
Using node 10.16.3 on a new installation, and empty `config.json` file, produces this error:

```log
[10/15/2019, 1:32:31 PM] Loading 0 accessories...
Load homebridge-bond.Bond
[10/15/2019, 1:32:31 PM] TypeError: Cannot read property 'bonds' of null
    at new BondPlatform (/usr/lib/node_modules/homebridge-bond/dist/index.js:24:20)
    at Server._loadDynamicPlatforms (/usr/lib/node_modules/homebridge/lib/server.js:358:30)
    at Server.run (/usr/lib/node_modules/homebridge/lib/server.js:92:8)
    at module.exports (/usr/lib/node_modules/homebridge/lib/cli.js:59:10)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

I can confirm this error goes away (and normal behavior exists) once the "bonds" key is added into "platforms" in `config.json`.

**Comment by iRayanKhan** [comment]
Can you please paste your config and the bond version of the plugin you are using?
I’m thinking it’s either a config typo or using the new config on an older version of the plugin.

**Comment by dxdc** [comment]
Actually, it has nothing to do with the bond version (I don't think), but I am running the latest beta if that is useful.

The problem (I believe) is that upon loading, it *assumes* config.json has a `bond` or `bonds` key in the config. If you have installed the plugin via npm, but didn't add it to `config.json` if fails with this error.

**Comment by dxdc** [comment]
I'll bet this is the problematic line @iRayanKhan :

`if (config.bonds === undefined) {`

It should be something like:

`if (config === undefined || config.bonds === undefined) {`

**Comment by iRayanKhan** [comment]
I mean't plugin version my bad, but you intrigued me at "beta", do you mean iOS beta?
V2.1.0 should have fixed it where if you didn't define the bond plugin in the config it gives a warning, but not a crash. I can make a fork to try out your suggestion, but I think we need some more info.
1) Plugin Version 
2) Config (Redact personal info)
3) Node version ``` node -v```

**Comment by dxdc** [comment]
@iRayanKhan 

- Node v10.16.3
- For `config.json`, you can actually try just deleting that file. That's the simplest way to reproduce it.
- From `homebridge-bond/package.json` homebridge-bond@2.3.0

Keep in mind that once I edited/added `config.json` it worked fine. My suggestion above I think will fix it -- you're looking for the `bonds` property on config, but part of that assumption is that `config` exists. The bond plugin is the only one I've seen that produces a fatal error in the absence of `config.json` file.

> I mean't plugin version my bad, but you intrigued me at "beta", do you mean iOS beta?

Yes, I'm running FW 2.9.1-beta using the latest bond-beta on iOS.

**Comment by iRayanKhan** [comment]
Interesting, I run over 10 instances of homebridge and it works fine. I'll make a backup of it and test it out once I get home. Some of my plugins such as Tplink-smarthome, and Tuya-Lan return a fatal error when their plugin in not defined, so I'll see what's different between this and those plugins and see if I can repro and find a root cause. I'm out rn so it'll be a bit for me before I can take a look at it.

**Comment by dxdc** [comment]
Thanks! Not urgent... just something to note if config file is missing.

**Comment by aarons22** [comment]
This is fixed in [2.3.1](https://github.com/aarons22/homebridge-bond/releases/tag/2.3.1)
